src/commands/misc/remind.ts

```diff
@@ -0,0 +1,161 @@
+import { ApplicationCommandOptionType } from "discord-api-types/v9";
+import { Client, CommandInteraction, DMChannel, TextChannel } from "discord.js";
+import { Pool } from "pg";
+
+import { Command } from "../type";
+
+const timeMatcher =
+  /^(?<weeks>\d+w)?(?<days>\d+d)?(?<hours>\d+h)?(?<minutes>\d+m)?(?<seconds>\d+s)?$/;
+
+async function createReminder(interaction: CommandInteraction, databasePool: Pool) {
+  const time = interaction.options.getString("when", true);
+  const reminderText = interaction.options.getString("reminder") || "Time's up!";
+  if (!timeMatcher.test(time) && time.toLowerCase() !== "rollover") {
+    interaction.reply({
+      content: 'You must supply a time to wait in the form of "1w2d3h4m5s" or "rollover".',
+      ephemeral: true,
+    });
+    return;
+  }
+  if (reminderText.length > 127) {
+    interaction.reply({
+      content: "Maximum reminder length is 128 characters.",
+      ephemeral: true,
+    });
+    return;
+  }
+  let timeToWait = 0;
+  let reminderTime = 0;
+  if (timeMatcher.test(time)) {
+    const timeMatch = timeMatcher.exec(time);
+    timeToWait =
+      7 * 24 * 60 * 60 * 1000 * parseInt(timeMatch?.groups?.weeks || "0") +
+      24 * 60 * 60 * 1000 * parseInt(timeMatch?.groups?.days || "0") +
+      60 * 60 * 1000 * parseInt(timeMatch?.groups?.hours || "0") +
+      60 * 1000 * parseInt(timeMatch?.groups?.minutes || "0") +
+      1000 * parseInt(timeMatch?.groups?.seconds || "0");
+    reminderTime = Date.now() + timeToWait;
+    await interaction.reply(`Okay, I'll remind you in ${time}.`);
```

Comment:
i think there is a small issue where there is an effective max on the amount of time the system can actually store for your reminder; when you or someone else did a 20 year reminder it just never got pushed to the db. i assume it is literally just due to 9007199254740991 int limit. so maybe a note here of like "Reminder not added; it was too far in the future, I encourage you to live dangerously in the present." if it's over the int limit?